forge/src/main/java/org/openjdk/skara/forge/github/GitHubRepository.java

```diff
@@ -261,14 +261,23 @@ public VCS repositoryType() {
     }
 
     @Override
-    public String fileContents(String filename, String ref) {
-        var conf = request.get("contents/" + filename)
-                          .param("ref", ref)
-                          .execute().asObject();
+    public Optional<String> fileContents(String filename, String ref) {
+        JSONValue content;
+        try {
+            content = request.get("contents/" + filename)
+                    .param("ref", ref)
+                    .execute();
+        } catch (UncheckedRestException e) {
+            // returns Optional.empty() on 404 File Not Found
+            if (e.getStatusCode() == 404 && e.getBody().contains("Not Found")) {
+                return Optional.empty();
+            }
+            throw e;
+        }
```

Comment:
Instead of catching this exception, please use the onError method on the request to register an error handler.

Reply:
Sorry, I couldn't find a good way to use errorHandler with the current `RestRequest#transformBadResponse`.
We will not be able to return `Optional.empty()` when the error is 404 Not Found.

Reply:
Ah, see your comments below. I know how to do it now. Thank you